### PR TITLE
feat! Create a script to build adaptor package artifacts

### PR DESCRIPTION
**Pull request by mwiebe**
* The breaking change is due to updating the adaptor runtime dependency to 0.4

### What was the problem/requirement? (What/Why)

We have designed a new structure for how we lay out application interface adaptors into packages. In order to simplify the packaging step, we'll put a script that generates an artifact with that layout into each adaptor git repo.

In order to perform the layout we wanted, I needed to modify the adaptor runtime. Those changes are in https://github.com/OpenJobDescription/openjd-adaptor-runtime-for-python/pull/69

The adaptor runtime dependency was out of date, After updating, it required changes from a socket_path to a server_path parameter.

### What was the solution? (How)

- Create a script to build adaptor packaging artifacts that follow the structure we designed
- Also deprecate the binary name NukeAdaptor, and add the name nuke-openjd
- Update the adaptor runtime dependency to 0.4, and fix up socket path into server path
- Update the submitter to provide both Rez and Conda packages metadata. The Rez package names are the same for now to keep backwards compatibility.

### What is the impact of this change?

The repo has a script that can generate a .tar archive containing a prefix layout of the adaptor python library and its dependencies, along with an entry point.

### How was this change tested?

The generated .tar archive was used to create a package, and that was tested on a Deadline Cloud farm.

### Did you run the "Job Bundle Output Tests"? If not, why not? If so, paste the test results here.
Yes.

```

Timestamp: 2024-02-09T13:02:36.991298-08:00
Running job bundle output test: D:\deadline-clients\deadline-cloud-for-nuke\job_bundle_output_tests\cwd-path

cwd-path
Test succeeded

Timestamp: 2024-02-09T13:02:38.047223-08:00
Running job bundle output test: D:\deadline-clients\deadline-cloud-for-nuke\job_bundle_output_tests\group-read

group-read
Test succeeded

Timestamp: 2024-02-09T13:02:38.939765-08:00
Running job bundle output test: D:\deadline-clients\deadline-cloud-for-nuke\job_bundle_output_tests\multi-load-save

multi-load-save
Test succeeded

Timestamp: 2024-02-09T13:02:40.019139-08:00
Running job bundle output test: D:\deadline-clients\deadline-cloud-for-nuke\job_bundle_output_tests\noise-saver

noise-saver
Test succeeded

Timestamp: 2024-02-09T13:02:41.064548-08:00
Running job bundle output test: D:\deadline-clients\deadline-cloud-for-nuke\job_bundle_output_tests\ocio

ocio
Test succeeded

All tests passed, ran 5 total.
Timestamp: 2024-02-09T13:02:44.283227-08:00
```

### Was this change documented?

No

### Is this a breaking change?

Yes, it updates the openjd adaptor runtime dependency which had a breaking change.

### CODE REVIEW

**Review by epmog — APPROVED** [review]

File: src/deadline/nuke_submitter/deadline_submitter_for_nuke.py [hunk 1]
```diff
@@ -204,21 +200,27 @@ def _get_parameter_values(
...
+        conda_param = {}
+        # Find the Packages parameter definition
         for param in queue_parameters:
             if param["name"] == "RezPackages":
                 rez_param = param
-                break
-        # Remove the deadline_cloud_for_nuke rez package
+            if param["name"] == "CondaPackages":
+                conda_param = param
+        # Remove the deadline_cloud_for_maya/maya-openjd package
```
Comment (jericht) [inline src/deadline/nuke_submitter/deadline_submitter_for_nuke.py]:
```suggestion
        # Remove the deadline_cloud_for_nuke/nuke-openjd package
```

File: src/deadline/nuke_submitter/deadline_submitter_for_nuke.py [hunk 2]
```diff
@@ -322,11 +324,18 @@ def on_create_job_bundle_callback(
 
     if not g_submitter_dialog:
         nuke_version = nuke.env["NukeVersionMajor"]
+        adaptor_version = ".".join(str(v) for v in adaptor_version_tuple[:2])
+
+        # Need Maya and the Maya OpenJD application interface adaptor
```
Comment (jericht) [inline src/deadline/nuke_submitter/deadline_submitter_for_nuke.py]:
```suggestion
        # Need Nuke and the Nuke OpenJD application interface adaptor
```

**Review by jericht — COMMENTED** [review]
LGTM! Just a couple typos or copy/paste misses

**Review by jericht — APPROVED** [review]

File: src/deadline/nuke_submitter/deadline_submitter_for_nuke.py (hunk 1 above)
Comment (mwiebe) [reply src/deadline/nuke_submitter/deadline_submitter_for_nuke.py]: Fixed, thanks for catching those!

**Review by epmog — APPROVED** [review]

**Pull request merged**